src/vdom/component.js

```diff
@@ -179,7 +196,11 @@ export function renderComponent(component, opts, mountAll, isChild) {
 		// flushMounts();
 
 		if (component.componentDidUpdate) {
-			component.componentDidUpdate(previousProps, previousState, previousContext);
+			try {
```

Comment:
try/catch is expensive, can we avoid doing this unless the component or an ancestor implements componentDidCatch?

Reply:
I believe the check would be more expensive because you have to check every ancestor of the component, which means the complexity goes up as your tree gets deeper. Its really a question of O(n) vs O(1), and with modern browsers, try/catch isn't even that expensive.

https://jsperf.com/try-catch-jsperf

Created a JSPerf to check, and it seems like the checking for  the existence of the method on just the component is more expensive then a standard try-catch

Reply:
👍 thanks for checking